### PR TITLE
Add HTMX loading styles and related documentation in styleguide

### PR DESCRIPTION
**Pull request by biyeun**
## Technical Summary
This PR adds new styles for HTMX loading states of:
- buttons
- checkboxes
- forms and their submit button

Documentation has been added to the HTMX and Alpine styleguide page which demos those common examples, and points out `hx-indicator` and `hx-disabled-elt` attributes to help with creating good loading states for the user.

Screenshot 2024-11-21 at 10 35 22 PM
Screenshot 2024-11-21 at 10 35 49 PM
Screenshot 2024-11-21 at 10 35 38 PM
Screenshot 2024-11-21 at 10 35 29 PM

### Safety story
Change to docs and to styling. very safe change

### Automated test coverage
no

### QA Plan
Not needed

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_htmx.scss [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+form.htmx-request {
+  .htmx-loading {
+    position: relative;
+    &::before {
+      content: " ";
+      @extend .spinner-border;
+      @extend .spinner-border-sm;
+      @include htmx-spinner(22px);
```
Comment (nospame) [inline corehq/apps/hqwebapp/static/hqwebapp/scss/commcarehq/_htmx.scss:17]: Given the various `px` sizing is this going to look right if it gets used on alternate sized inputs e.g. `btn-sm` or `btn-lg`? Not that we do it often, just wondering out loud.

File: corehq/apps/styleguide/templates/styleguide/bootstrap5/htmx_and_alpine.html [hunk 2]
```diff
@@ -1,4 +1,9 @@
 {% extends 'styleguide/bootstrap5/base.html' %}
+{% load hq_shared_tags %}
+
+{% block head %}
+  <script src="{% static 'htmx.org/dist/htmx.min.js' %}"></script>
```
Comment (nospame) [inline corehq/apps/styleguide/templates/styleguide/bootstrap5/htmx_and_alpine.html:5]: Was this loaded somewhere else prior to this change?

File: corehq/apps/styleguide/templates/styleguide/bootstrap5/examples/htmx_loading_form.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<form
+  hx-get="{% url 'styleguide_a_hanging_view' %}"
+  hx-disabled-elt="find button"
+  hx-swap="none"  {# example view returns nothing to display #}
+>
+  <div class="mb-3">
+    <label for="id_value" class="form-label">Value</label>
+    <input type="text" name="value" class="textinput form-control" id="id_value"/>
```
Comment (nospame) [inline corehq/apps/styleguide/templates/styleguide/bootstrap5/examples/htmx_loading_form.html:8]: nit: should this be `id-value` for an HTML styled id?

File: corehq/apps/styleguide/templates/styleguide/bootstrap5/examples/htmx_loading_checkbox.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<div class="form-check">
+  <input
+    class="form-check-input" type="checkbox" value="" id="flexCheckDefault"
```
Comment (nospame) [inline corehq/apps/styleguide/templates/styleguide/bootstrap5/examples/htmx_loading_checkbox.html:3]: nit: should this be `flex-check-default` for an HTML styled ID?

**Review by nospame — COMMENTED** [review]
Couple of small questions, like the way this looks though.

**Review by jingcheng16 — APPROVED** [review]
Reviewed together with @mjriley in group review.

**Pull request merged**